utils/errors/err_group.go

```diff
@@ -0,0 +1,24 @@
+package errors
+
+import (
+	"github.com/dymensionxyz/dymint/types"
+	"golang.org/x/sync/errgroup"
+)
+
+/*
+TODO: move to sdk-utils
+*/
+
+// ErrGroupGoLog calls eg.Go on the errgroup but it will log the error immediately when it occurs
+// instead of waiting for all goroutines in the group to finish first. This has the advantage of making sure all
+// errors are logged, not just the first one, and it is more immediate. Also, it is guaranteed, in case that
+// of the goroutines is not properly context aware.
+func ErrGroupGoLog(eg *errgroup.Group, logger types.Logger, fn func() error) {
```

Comment:
this is nice, but what if we want to include some extra information in the log message, like those that we can only get locally in the `fn`'s body?

Reply:
cross that bridge when we get to it